### PR TITLE
Update introduction to python documentation

### PR DESCRIPTION
**Pull request by StephenSmith25**
**Description of work.**

This PR makes a few adjustments to the introduction to python course, namely:

- Removes a reference to python 2, which is no longer relevant
- Updates the file handling to use the ``with`` statement.
- A small example of using a nametuple, as it is a useful collection due to the clarity of the contents it provides.

**To test:**
 - Builds pass
 - Build the documentation and check everything looks correct.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: docs/source/tutorials/introduction_to_python/working_with_files.rst [hunk 1]
```diff
@@ -78,26 +89,24 @@ way of processing a file.
...
-   file = open('MyFile.txt')
-   for line in file:
-       line = line.rstrip()
-       print(line)
-   file.close()
+   with open('MyFile.txt') as file:
+       for line in file:
+           line = line.rstrip()
+           print(line)
+       file.close()
```
Comment (Harrietbrown) [inline docs/source/tutorials/introduction_to_python/working_with_files.rst:109]: You don't need to close the file when using with.

**Review by Harrietbrown — CHANGES_REQUESTED** [review]
just thins one thing and it will be good to go.

File: docs/source/tutorials/introduction_to_python/solutions_to_exercises.rst [hunk 2]
```diff
@@ -19,14 +19,6 @@ Exercise 1
...
 
-   # range(i,j) produces a list of numbers from i -> j-1
-   #python 2
-   for i in range(1,21):
-       print str(i*i).center(3), # center is a function for strings that centers the contents to the given width
-       if i % 4 == 0:
-           print
-
-   #Python 3
    #for i in range(1,21):
```
Comment (Harrietbrown) [inline docs/source/tutorials/introduction_to_python/solutions_to_exercises.rst:22]: This should also be uncommented.

Comment (StephenSmith25) [reply docs/source/tutorials/introduction_to_python/solutions_to_exercises.rst]: Ah yes missed this, thank you 👍

**Review by Harrietbrown — APPROVED** [review]
Good Job, the example code is correct as expected.

**Pull request merged**